collect/util.go

```diff
@@ -39,25 +40,59 @@ func resolveRootOwner(kc client.Client, obj client.Object) (*inventory.RootOwner
 			Name:       rootObj.GetName(),
 			Namespace:  rootObj.GetNamespace(),
 		}
-		return rootOwner, nil
+
+		meta := metav1.ObjectMeta{
+			Name:              rootObj.GetName(),
+			Namespace:         rootObj.GetNamespace(),
+			Labels:            rootObj.GetLabels(),
+			Annotations:       rootObj.GetAnnotations(),
+			CreationTimestamp: rootObj.GetCreationTimestamp(),
+			OwnerReferences:   rootObj.GetOwnerReferences(),
+		}
+
+		apiGroup := rootOwner.APIGroup
+		if apiGroup == "" {
+			apiGroup = "core"
+		}
+
+		mapping, err := kc.RESTMapper().RESTMapping(rootObj.GroupVersionKind().GroupKind(), rootObj.GroupVersionKind().Version)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ownerWorkload := &inventory.Workload{
+			TypeMeta: inventory.TypeMeta{
+				Kind:         rootOwner.Kind,
+				APIGroup:     apiGroup,
+				APIVersion:   rootOwner.APIVersion,
+				ResourceType: mapping.Resource.Resource,
```

Comment:
Is this (Resource.Resource) intentional?

Reply:
Yeah, for some reason they "hide" the resource type quite deep in the hierarchy.